Objects/genericaliasobject.c

```diff
@@ -430,17 +467,39 @@ _Py_subs_parameters(PyObject *self, PyObject *args, PyObject *parameters, PyObje
             jarg++;
             continue;
         }
-
+        // Recursively substitute params in lists/tuples.
+        if (PyTuple_Check(arg) || PyList_Check(arg)) {
+            PyObject *subargs = _Py_subs_parameters(self, arg, parameters, item);
```

Comment:
Passing `parameters` as is here seems wrong; what if we only need part of the parameters? `_Py_subs_parameters` will error if the count doesn't match. The test case I suggested should help verify this.

Reply:
This actually uncovered an issue where nested args tuples would be converted to lists, which should be now fixed.
Also added more tests if you feel like having another look :)